apisix/plugins/openwhisk.lua

```diff
@@ -104,8 +108,31 @@ function _M.access(conf, ctx)
         return 503
     end
 
+    -- parse OpenWhisk JSON response
+    -- OpenWhisk supports two types of responses, the user can return only
+    -- the response body, or set the status code and header.
+    local result, err = core.json.decode(res.body)
+
+    if not result or err then
+        core.log.error("failed to parse openwhisk response data: ", err)
+        return 503
+    end
+
     -- setting response headers
-    core.response.set_header(res.headers)
+    if result.headers ~= nil then
+        core.response.set_header(result.headers)
+    end
+
+    if result.statusCode ~= nil and result.body ~= nil then
```

Comment:
we can use `local code = result.statusCode` and `local body = result.body`, and then use `code` and `body`

Reply:
resolved.